inf-clojure.el

```diff
@@ -843,6 +847,115 @@ HOST is the host the process is running on, PORT is where it's listening."
   (interactive "shost: \nnport: ")
   (inf-clojure (cons host port)))
 
+(defvar-local inf-clojure-socket-callback nil
+  "Used to transfer state between the socket process buffer & REPL buffer.")
+
+(defvar-local inf-clojure-socket-buffer nil
+  "Used to kill the associated socket buffer when it's REPL buffer is killed.")
+
+(defun inf-clojure-socket-filter (process output)
+  "A filter that gets triggered each time the socket receives new OUTPUT.
+This function prints out the output received but also
+watches for a prompt using the `inf-clojure-prompt' regexp, once
+this happens a callback is triggered if available.  The callback
+is intended to be used to trigger a `inf-clojure-connect' once we
+can determine that a socket REPL is ready to receive a
+connection.
+
+PROCESS is the process object that is being filtered.
+
+OUTPUT is the latest data received from the process"
+  (let ((server-buffer (process-buffer process)))
+    (when (buffer-live-p server-buffer)
+      (with-current-buffer server-buffer
+        (insert output)))
+    (let ((prompt-displayed (string-match inf-clojure-prompt output)))
+      (when prompt-displayed
+	(message (format "REPL prompt found for %s" (process-name process)))
+	(with-current-buffer server-buffer
+	  (when inf-clojure-socket-callback
+	    (funcall inf-clojure-socket-callback)))))))
+
+(defun inf-clojure-repl-sentinel (process event)
+  "Ensures socket REPL are cleaned up when the REPL buffer is closed.
+
+PROCESS is the process object that is connected to a socket REPL.
+
+EVENT is the event that triggered this function to be called."
+  (when (not (process-live-p process))
+    (let ((repl-buffer (process-buffer process)))
+      (with-current-buffer repl-buffer
+        (when inf-clojure-socket-buffer
+          (kill-buffer inf-clojure-socket-buffer))))))
+
+(defvar inf-clojure-socket-repl-startup-forms
+  '((lein . "JVM_OPTS='-Dclojure.server.repl={:port %s :accept clojure.core.server/repl}' lein repl")
+    (boot . "export BOOT_JVM_OPTIONS='-Dclojure.server.repl=\"{:port %s :accept clojure.core.server/repl}\"' boot repl")
+    (clojure . "clojure -J-Dclojure.server.repl=\"{:port %s :accept clojure.core.server/repl}\"")
+    (cljs . "clojure -J-Dclojure.server.repl=\"{:port %s :accept cljs.server.browser/repl}\"")
+    (lein-clr . "JVM_OPTS='-Dclojure.server.repl={:port %s :accept clojure.core.server/repl}' lein clr repl")
+    (planck . "planck -n %s")
+    (babashka . "bb socket-repl %s")))
+
+(defcustom inf-clojure-socket-repl-port
+  nil
+  "Port to be used when creating a socket REPL via `inf-clojure-socket-repl'.
+If left as nil a random port will be selected between 5500-6000."
+  :type '(choice integer (const nil)))
```

Comment:
this should also have a `:package-version` property.

Reply:
added this as version `3.2.1`